clients/src/main/java/org/apache/kafka/common/security/kerberos/KerberosLogin.java

```diff
@@ -92,14 +92,14 @@
      * @throws javax.security.auth.login.LoginException
      *               Thrown if authentication fails.
      */
-    public void configure(Map<String, ?> configs, final String loginContextName) {
-        super.configure(configs, loginContextName);
+    public void configure(Map<String, ?> configs, Configuration jaasConfig, final String loginContextName) {
```

Comment:
In `reLogin` we have `loginContext = new LoginContext(loginContextName, subject);`. Shouldn't we be passing the `jaasConfig` there? Also, I'm a bit unclear on why we pass a `subject` since we didn't in the initial construction.